examples/FileIO/ReadBytesFromFile.dfy

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT
+*******************************************************************************/
+
+// RUN: %dafny /compile:0 "%s"
+// RUN: echo -n > "%t"
```

Comment:
Why is this needed?

Reply:
The `"%t"` temporary file must be truncated each time lit processes this file, or else subsequent `lit` invocations will lead to accumulating output there. (This really only applies locally, since we don't re-run tests in CI.)

* In `dafny-lang/dafny` we usually write `RUN: %dafny /compile:0 "%s" > "%t"` so we can diff against it, but that command's output isn't deterministic when run with the default `verificationLogger` parameters (which I don't see fit to disable here).
* We could change the first `RUN: %baredafny run ...` command to use `> "%t"` instead of `>> "%t"`, but I don't like having that difference between lines (in case someone reorders/removes lines, the truncation would break).
* So instead it seems easiest to just explicitly truncate the file with an `echo` (which lit handles as a [builtin command + redirection](https://github.com/llvm/llvm-project/blob/86fe4dfdb626d472f88bf8331da7cc98e9631cfe/llvm/utils/lit/lit/TestRunner.py#L317), anyway).

Reply:
Ah interesting! I usually think of `%t` as a fresh temporary random value, which isn't true. In that case I agree this approach is a bit cleaner and more maintainable.

Although now that I look closer at this, I don't think there's any value in checking the output of these commands at all. All you care about is that the commands succeed. It's the same reason `%testDafnyForEachCompiler` over in `dafny-lang/dafny` doesn't look at the output of the initial verification call.

Reply:
Yeah, I agree. The `.expect` file doesn't have any interesting content.

Reply:
I suppose that's true, I've removed them.